### ISSUE TITLE
vlv index may hang in bdb_remove

### ISSUE THREAD

**Issue by progier389** [389ds/389-ds-base#6029]
**Issue Description**
Server may hangs within vlv index if operations on that vlv index were previously done. 
Redoing the reindex after restarting the server fix the issue.

**Package Version and Platform:**
 - Platform: Fedora
 - Package and version: main branch 

**Steps to Reproduce**
1. Edit regression_test.py and comment M2.restart() in test_vlv_cache_subdb_names
2. Comment/remove the workaround for this issue:
```
if get_default_db_lib() == "bdb":
        M2.restart()
```
4. Run ( pytest  $PWD -k 'test_bulk_import_when_the_backend_with_vlv_was_recreated or test_vlv_cache_subdb_names')
5. See error: The test hangs. a pstack on supplier2 shows that bdb_remove waits on some db locks.

**Expected results**
Command should complete

**Additional context**
Found when developing a CI test, set low priority because bdb is slowly getting deprecated and so far nobody reported the issue. (so its looks like a corner case)
